base/channels.jl

```diff
@@ -454,6 +499,144 @@ function put_unbuffered(c::Channel, v)
     return v
 end
 
+"""
+    append!(c::Channel, iter)
+
+Append all items in `iter` to the channel `c`. If the channel is buffered, this operation requires
+fewer `lock` operations than individual `put!`s. If the length of the iterator
+is greater than the available buffer space, the operation will block until enough elements
+are taken from the channel to make space for the new elements.
+
+# Examples
+
+```jldoctest
+julia> c = Channel(3);
+
+julia> append!(c, 1:3);
+
+julia> take!(c)
+1
+
+julia> take!(c)
+2
+
+julia> take!(c)
+3
+```
+
+!!! compat "Julia 1.12"
+    Requires at least Julia 1.12.
+"""
+function append!(c::Channel, iter::T) where {T}
+    if IteratorSize(iter) isa Union{HasLength, HasShape}
+        len = length(iter)
+        # short circuit for small iters
+        if len == 0
+            return c
+        elseif len == 1
+            put!(c, @inbounds vec[begin])
+            return c
+        end
+    end
+
+    return isbuffered(c) ? append_buffered(c, iter) : append_unbuffered(c, iter)
+end
+function append!(c_dst::Channel, c_src::Channel{T}) where {T}
```

Comment:
might be worth checking that `c_dst !== c_src` otherwise I think `append!` will actually empty the channel